spec/facility_spec.rb

```diff
@@ -1,26 +1,132 @@
+      expect(@facility_1.services).to eq([])
+      @facility_1.add_service('New Drivers License')
+      @facility_1.add_service('Renew Drivers License')
+      @facility_1.add_service('Vehicle Registration')
+      expect(@facility_1.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+    end
+  end
+
+  describe '#register_vehicle' do
+    it 'can register a vehilce' do
+      @facility_1.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      registered_vehicles = @facility_1.register_vehicle(@cruz)
+      expect(registered_vehicles.first).to be_a Vehicle
+      expect(@cruz.registration_date).to eq(Date.today) 
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.registered_vehicles).to include(@cruz)
+      expect(@facility_1.collected_fees).to eq 100 
+    end
+
+    it 'can register multiple vehicles' do
+      @facility_1.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+      @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+      @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+      registered_vehicles = @facility_1.register_vehicle(@cruz)
+      registered_vehicles = @facility_1.register_vehicle(@camaro)
+      registered_vehicles = @facility_1.register_vehicle(@bolt)
+      expect(@cruz.registration_date).to eq(Date.today) 
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to include(@cruz, @camaro, @bolt)
+      expect(@facility_1.collected_fees).to eq 325
+    end
+  end
+
+  describe '#administer_written_test' do
+    it 'can administer the written test' do
+     facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+     registrant_1 = Registrant.new('Bruce', 18, true)
+
+     facility_1.add_service('Written Test')
+     expect(facility_1.administer_written_test(registrant_1)).to be true
+     expect(registrant_1.license_data[:written]).to be true
+    end
+
+    it 'fails written test if no service is available' do
+      facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      registrant_1 = Registrant.new('Bruce', 18, true)
+      expect(facility_1.administer_written_test(registrant_1)).to be false
+      expect(registrant_1.license_data[:written]).to be false
+    end
+  end
+
+  describe '#administer_road_test' do
```

Comment:
This should also test that a road test is not administered if the registrant has not completed their written test.